packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ChannelAwareNotificationBuilder.java

```diff
@@ -39,6 +43,16 @@ protected String getChannelId() {
       return null;
     }
 
+    NotificationTrigger trigger = getTrigger();
+    if (!(trigger instanceof FirebaseNotificationTrigger)) {
+      return getFallbackNotificationChannel().getId();
+    }
+
+    FirebaseNotificationTrigger firebaseNotificationTrigger = (FirebaseNotificationTrigger) trigger;
+    if (firebaseNotificationTrigger.getRemoteMessage().getData().containsKey("channelId")) {
+      return firebaseNotificationTrigger.getRemoteMessage().getData().get("channelId");
```

Comment:
I think this logic should in fact belong to NotificationTrigger.

I would see it like that: every notification in fact needs channel (for Android Oreo+). we might hence add `getChannelId` to NotificationTrigger and provide default which returns null. Here in the Builder, we would only check it for nullability and provide fallback channel if needed.